### ISSUE TITLE
Slider Preferences do not save.

### ISSUE THREAD

**Issue by Skrappjaw** [jack-mixer/jack_mixer#164]
Mixer preferences do not save. Interface, Scales, nor MIDI. I like the 70DB scale to avoid clipping channels, and the Pickup Value option avoids slamming a fader to the gills on accident. But every time I reload the preset, these are set to defaults.

**Comment by SpotlightKid** [comment]
Thanks for the report. What version of jack_mixer are you using and which Linux distro?  Can you run jack_mixer from a terminal with the verbose flag and load your settings and post the terminal output here? Also, please post your mixer settings and your global settings file (in ~/.config/jack_mixer).

**Comment by Skrappjaw** [comment]
Fedora 35
Version 14

> $ jack_mixer -d -c /home/skrapp/Audio_configs/UMC-Korg 
> WARNING: Cannot load PyXDG. Your preferences will not be preserved across jack_mixer invocations

There is no folder in .config for jack_mixer

> <?xml version="1.0" ?>
> <jack_mixer geometry="755x435" paned_position="528" visible="True">
> 	<input_channel name="Mic" type="mono" volume="-15.826771" balance="0.000000" wide="True" out_mute="False" volume_midi_cc="1" mute_midi_cc="48" solo_midi_cc="32"/>
> 	<input_channel name="Guitar" type="mono" volume="-inf" balance="0.000000" wide="True" out_mute="True" volume_midi_cc="2" mute_midi_cc="49" solo_midi_cc="33"/>
> 	<input_channel name="SYSTEM-AUDIO" type="stereo" volume="2.598425" balance="0.000000" wide="True" out_mute="False" volume_midi_cc="3" mute_midi_cc="50" solo_midi_cc="34"/>
> 	<input_channel name="Voip" type="stereo" volume="-3.070866" balance="0.000000" wide="True" out_mute="False" volume_midi_cc="4" balance_midi_cc="24" mute_midi_cc="51" solo_midi_cc="35"/>
> 	<input_channel name="Games" type="stereo" volume="-8.740157" balance="0.000000" wide="True" out_mute="False" volume_midi_cc="5" balance_midi_cc="28" mute_midi_cc="52" solo_midi_cc="36"/>
> 	<output_channel name="Voip_out" type="stereo" muted_channels="SYSTEM-AUDIO|Voip" color="rgb(38,162,105)" volume="-9.212599" balance="0.000000" wide="True" out_mute="False" volume_midi_cc="7" balance_midi_cc="0" mute_midi_cc="54"/>
> 	<output_channel name="Main" type="stereo" color="rgb(153,193,241)" volume="2.125984" balance="0.000000" wide="True" out_mute="False" volume_midi_cc="8" mute_midi_cc="55"/>
> 	<gui_factory confirm-quit="False" default_meter_scale="K20" default_slider_scale="linear_30dB" midi_behavior_mode="0" use_custom_widgets="False" vumeter_color="#ccb300" vumeter_color_scheme="default"/>
> </jack_mixer>

**Comment by SpotlightKid** [comment]
Release 14 is more than two years old. Current release is version 17, released more than a year ago. Please test with a current version.

Also, as the warning message says, you need PyXDG installed for proper settings saving and restoring.

**Comment by Skrappjaw** [comment]
Makes sense. Must be a packaging issue on Fedora's end.

**Comment by Skrappjaw** [comment]
Is there a way to compile this without jack installed? I'm on pipewire and have no intention of "jacking" up my system trying to compile one program just to revert back to pipewire.

**Comment by SpotlightKid** [comment]
Tbh, I don't really know. I don't use pipewire's JACK part, only the pulseaudio replacement.

On Arch Linux, the package `pipewire-jack` installs Pipepwire's version of `libjack.so` in the usual place (`/usr/lib`), so I guess there compilation should work out-of-the-box, if that package is installed and it reports a version that is compatible with that of JACK1 or JACK2. I haven't tried, though. I'm also not sure, how Pipewire is packaged on Fedora and where it installs its version of `libjack.so`.

**Comment by Skrappjaw** [comment]
I found the pipewire-jack files in a different location. Tried to make a simlink in /usr/lib but it shows not a recent enough version of jack. But Fedora has version 1.9.19. Compiler only requires 1.9.11.

**Comment by Skrappjaw** [comment]
Noting. Did a search on pkgs.org through the all the jack/pipewire packages fedora ships.
Found it. It's not shipped in the default packages, but it is in the devel packages. Add this note for Fedora users to the installation instructions. I'll also attempt to hit up the Fedora team to let them know jack_mixer is severely out of date/version parity. Thanks for the help.
`sudo dnf install pipewire-jack-audio-connection-kit-devel
`

**Comment by SpotlightKid** [comment]
Yes, on debian and redhat-based distros you always need to install the -dev packages for the libraries you want to compile/link with.

Thanks for your research.
